main.py

```diff
@@ -257,16 +245,25 @@ def _kill_tb_child():
     atexit.register(_kill_tb_child)
 
 
-def get_best_checkpoint_path(run_dir, phase):
+def get_best_checkpoint_path(run_dir, phase, task_name=None):
     """ Look in run_dir for model checkpoint to load.
     Hierarchy is
-        1) best checkpoint from the phase so far
-        2) if we do only target training without pretraining, then load checkpoint before
+        1) best task-specific checkpoint for target_train, used when evaluating
```

Comment:
Only when evaluating, or also when doing target-task training?

Reply:
We should make sure *not* to restore a partially-completed target_task training run unless we're sure that that works, and we restore things like step count and optimizer state appropriately.

Reply:
This is only used during evaluation, and is a retrieval type function. During target task training, we store the parameters at each epoch for each task being target trained, which is retrieved here.

Reply:
SGTM!